### ISSUE TITLE
HIGH-PRIORITY: Static Compilation

### ISSUE THREAD

**Issue by stugol** [ozra/onyx-lang#80]
[Placeholder issue to ensure that static compilation remains an active high-priority goal]

Crystal does not support static (no dependancies) compilation. Onyx needs to support this. Presumably (due to OS limitations) `libc` will continue to be dynamically dependant, but everything else _must_ be statically linkable. In fact, I propose that static linking be the **_default mode!_**

In this day and age, there is no advantage to dynamic linking; and plenty of disadvantages.

**Comment by ozra** [comment]
dev-build: default to dynamic, release-build: default to static. Much reasonable.

**Comment by ozra** [comment]
I took the liberty to edit out "HIGH PRIORITY: " from title - prioritization is in Roadmap issue.

**Comment by Sod-Almighty** [comment]
Continued lack of this feature is the single biggest reason why I have not yet written a single line of Onyx. Once this is solved, I will start actively using it.

**Comment by ozra** [comment]
Yes, this is highly desirable! Effort must however go into actually shaping the language so that the loose ends are cut off and the specs stabilize to a predictably usable form (this and that are obviously still changing in rather profound ways). Spending time on architecture level compiler coding must _unfortunately_ come later for me - I only have two hands. Help is highly appreciated! B-)

**Comment by Sod-Almighty** [comment]
I'm afraid I would be utterly useless working on the compiler. I can probably help with other stuff though, if there's a need for it. What needs doing besides the compiler?

**Comment by ozra** [comment]
Well, documentation is an obvious candidate. I could focus on finishing the last "core" parts there, so the repo can go public. But first finish the type introspection. Then I'll go for braces after those two.

**Comment by Sod-Almighty** [comment]
What needs doing on the documentation? Are we starting from scratch, or tidying up an existing document?

Are we going for a language reference and a brief tutorial? A little bit like the Crystal doc page?

**Comment by Sod-Almighty** [comment]
I guess I could just use the Crystal docs as a starting point. What format are we after? A PDF? A bunch of HTML files in the repo? An actual website?

**Comment by ozra** [comment]
Continued in #99

**Comment by ozra** [comment]
Continued in #99

**Comment by ozra** [comment]
Took the liberty of changing the title, since it's linking we're talking about. Static compilation brings the mind to `constexpr`-stuff for me ;-)
